week-8/Homework/mandatory/3-dog-photo-gallery/script.js

```diff
@@ -0,0 +1,17 @@
+const button = document.querySelector("#button")
+const dogList = document.querySelector('#dogList')
+
+button.addEventListener('click', () => {
+    fetch("https://dog.ceo/api/breeds/image/random")
+    .then( data => data.json())
+    .then( data => {
+        const img = document.createElement('img')
+        img.src = data.message
+        const li= document.createElement('li')
+        dogList.appendChild(li)
+        li.appendChild(img)  
+
+    }) 
+    .catch( err => console.log(err))
```

Comment:
This works really well for showing the developer the error information, but it doesn't really tell a normal user that anything has gone wrong. Can you think how to also let the user know what's happening / what to expect? Similarly for the humour question :)